blog/2019-10-02-MigratingOptaWebEmployeeRosteringToSpringBoot.adoc

```diff
@@ -0,0 +1,195 @@
+----
+public static final String INITIAL_DATA_PROPERTY = "optaweb.generator.initial.data";
+
+public enum InitialData {
+    EMPTY,
+    DEMO_DATA // default
+}
+----
+
+In `RosterGenerator.java`, we use the system property to determine whether to generate data or not.
+Notice that if `initialData = EMPTY`, the method returns without generating any data:
+
+[source,java]
+----
+SystemPropertiesRetriever.InitialData initialData = SystemPropertiesRetriever.determineInitialData();
+
+switch (initialData) {
+    case EMPTY:
+        return;
+    case DEMO_DATA:
+        //Generate demo data
+}
+----
+
+To configure the system properties and start with an empty database, add this to the `applications.properties` file:
+
+[source,properties]
+----
+optaweb.generator.initial.data=EMPTY
+----
+
+This sets the `optaweb.generator.initial.data` system property for the application when it starts.
+
+== REST API
+
+Now let's actually interact with the backend application.
+An easy way to test the REST API is to use the Swagger framework, which OptaWeb Employee Rostering supports.
+
+As an example, here are the `Tenant` REST methods shown in the Swagger user interface:
+
+image::tenantMethods.png[]
+
+As we can see, each method has documentation, the relative path URL, and its corresponding HTTP request.
+The Swagger UI makes it very easy to test any method in the backend.
+Let's say we wanted a list of all the Tenants in the application.
+We start by selecting the corresponding method to display the method details:
+
+image::getTenantsMethod.png[]
+
+Here, the `Example Value` field shows a sample value of a Tenant list, serialized to JSON format.
+We see that each Tenant contains the `id`, `name`, and `version` fields, which are saved in the database whenever a Tenant is added.
+Also note the `Try it out` button in the top right corner.
+We can use it to send HTTP requests with ease for testing purposes.
+Let's try it out:
+
+image::executeGetTenants.png[]
+
+An `Execute` button will appear, and clicking on it sends a GET request that gets all the saved Tenants in JSON format.
+Notice the GET request is sent to the `Request URL`, and the response is found in the `Response body`.
+The server returns the `200` code, which means the request was handled with no issues.
+
+Nice, we just sent our first HTTP request!
+But the Employee Rostering application does much more than just getting entities.
+There are methods for creating, deleting, and updating various entities (remember the Domain Model above?), which the application uses to modify the database, as well as methods that interact with the OptaPlanner API.
+
+Let's try a more interesting method:
+
+image::solveMethod.png[]
+
+This method is the core of the OptaWeb Employee Rostering application.
+It starts the OptaPlanner solver that assigns work shifts to employees for a specific Tenant.
+The solver uses heuristics and algorithms to find the most optimal solution under time and resource constraints.
```

Comment:
Good place to link to Optaplanner documentation for local search, I think.

Reply:
Linking into sections of the optaplanner docs is maintenance unfriendly, unfortunately. (It shouldn't be, but it is what it is.)
You can use the "latest" link to avoid hard coding a version number, for example https://docs.optaplanner.org/latestFinal/optaplanner-docs/html_single/  but I generally avoid linking into the user guide for blogs.